lib/p2p/Pool.ts

```diff
@@ -92,6 +92,20 @@ class Pool extends EventEmitter {
     this.peers.add(peer);
   }
 
+  public disconnectPeer = async (address: string, port: number): Promise<void> => {
+    const socketAddress = new SocketAddress(address, port);
+    if (this.peers.has(socketAddress)) {
+      const peer = this.peers[socketAddress.toString()];
```

Comment:
This line assignment doesn't return a peer. 

You need to implement `get` on `PeerList`, and after doing so you can avoid using `has`.

Reply:
My preferred approach would be to make `remove` accept a `SocketAddress` and return the `Peer` that it removed, or null if no peer with that address exists.

Reply:
I used solely `get` because after you do `peer.destroy`, it will trigger an event that will remove it from `peerList` anyway. 

But I implemented `remove` as you suggested, so we'll have it.